src/util.js

```diff
@@ -130,6 +130,82 @@ export function mapValues(object, mapper) {
   )));
 }
 
+/**
+ * Formats date and the time
+ *
+ * @param {string} text The text to be formatted
+ * @returns {string} Returns formatted text
+ */
+export function dateTimeFormatting(text) {
+  const locale = 'en-us';
+  const formatOptions = {year: 'numeric', day: 'numeric'};
+
+  const dateRegex = /{{DATE\(\d{4}-\d{2}-\d{2}T?\d{2}:\d{2}:\d{2}Z?([-+]{1}\d{2}:\d{2})?,\s*(SHORT|LONG|COMPACT)\)}}/g;
+  const timeRegex = /{{TIME\(\d{4}-\d{2}-\d{2}T?\d{2}:\d{2}:\d{2}Z?([-+]{1}\d{2}:\d{2})?\)}}/g;
+  const dayRegex = /,(\s{1}.*\s\d{1,2}),/g;
+
+  const dateFormats = {
+    SHORT: 'short',
+    LONG: 'long',
+    COMPACT: 'narrow',
+  };
+
+  function replaceDate(dateMatch, _, type) {
+    const format = dateFormats[type] || dateFormats.COMPACT;
+    const options = {
+      ...formatOptions,
+      weekday: format === dateFormats.COMPACT ? undefined : format,
+      month: format === dateFormats.COMPACT ? 'numeric' : format,
+    };
+    const cleanerRegex = /\(.*,/g;
+    const dateString = dateMatch.match(cleanerRegex)[0].slice(1, -1);
```

Comment:
if you need the date string from the text, replaceAll will give it to you if you put parens around the date in the regex:
```js
'{{DATE(2017-02-14T06:00:00Z, SHORT)}}'
  .replaceAll(
    /{{DATE\((\d{4}-\d{2}-\d{2}T?\d{2}:\d{2}:\d{2}Z?(?:[-+]{1}\d{2}:\d{2})?),?\s*(SHORT|LONG|COMPACT)?\)}}/g,
    (_, date, format) => { console.log('date:', date, 'format:', format); }
  );
VM2018:4 date: 2017-02-14T06:00:00Z format: SHORT